infrastructure/local-provisioners/azurerm_apim.ts

```diff
@@ -0,0 +1,510 @@
+      const contentValue = `${backendUrl}${apiEntry.api.specsPath}?code=${
+        masterKey
+      }`;
+
+      winston.info(
+        `Adding API from URL: ${backendUrl}${apiEntry.api.specsPath}`
+      );
+
+      // Add API to API management
+      await apiClient.api.createOrUpdate(
+        config.azurerm_resource_group,
+        config.azurerm_apim,
+        apiEntry.id,
+        {
+          contentFormat: "swagger-link-json",
+          contentValue,
+          displayName: apiEntry.api.displayName,
+          path: apiEntry.api.path,
+          protocols: ["https"],
+          // WARNING: serviceUrl is taken from the swagger specs "host" field
+          // and there's no way to override that value here: it *must* be changed
+          // manually in the API management settings
+          // (or provide a real value in the swagger specs).
+          serviceUrl: backendUrl
+        }
+      );
+      // Add API to products
+      if (includeProducts) {
+        await Promise.all(
+          apiEntry.products.map(async (product: string) => {
+            winston.info(
+              `Import API product into the API management: ${product}`
+            );
+            return apiClient.productApi.createOrUpdate(
+              config.azurerm_resource_group,
+              config.azurerm_apim,
+              product,
+              apiEntry.id
+            );
+          })
+        );
+      }
+      // Add a policy to the API reading it from a file
+      if (includePolicies && apiEntry.policyFile) {
+        winston.info(
+          `Import API policy into the API management: ${apiEntry.policyFile}`
+        );
+        const policyContent = fs.readFileSync(
+          path.join(__dirname, "..", "api-policies", apiEntry.policyFile),
+          "utf8"
+        );
+        await apiClient.apiPolicy.createOrUpdate(
+          config.azurerm_resource_group,
+          config.azurerm_apim,
+          apiEntry.id,
+          {
+            policyContent
+          },
+          // If-Match: *
+          "*"
+        );
+      }
+    })
+  );
+};
+
+/**
+ * Setup ADB2C authentication for developer portal users.
+ */
+const setupAdb2c = (
+  apiClient: apiManagementClient,
+  config: IResourcesConfiguration,
+  adb2cTenantId: string,
+  adb2cClientId: string,
+  adb2cClientSecret: string
+) => {
+  return apiClient.identityProvider.createOrUpdate(
+    config.azurerm_resource_group,
+    config.azurerm_apim,
+    "aadB2C",
+    {
+      allowedTenants: [adb2cTenantId],
+      clientId: adb2cClientId,
+      clientSecret: adb2cClientSecret,
+      identityProviderContractType: "aadB2C",
+      signinPolicyName: config.azurerm_adb2c_policy,
+      signupPolicyName: config.azurerm_adb2c_policy
+    }
+  );
+};
+
+const setupLogger = async (
+  apiClient: apiManagementClient,
+  config: IResourcesConfiguration,
+  eventHubConnectionString: string
+) => {
+  winston.info("Create an EventHub logger for the API management");
```

Comment:
I think here we need a separated "EventHub logger" provisioner and resource, provisioners need to be scoped on a single Azure resource

Reply:
ok

Reply:
just want to point out that it is not a new resources, the task just set a configuration parameter in the API settings